### PR TITLE
Insert form

### PR DESCRIPTION
**Pull request by manandearth**
The route "/invoices-insert" POST returns a param-map that needs coercing (is that the term?) to keywords (i.e `{"amount" "12"} -> {:amount 12} `). 
I saw it done with `json-read` using a `:key-fn` `keyword`  but with the current interceptors situation I am not sure how can this be applied.
All this assuming that I am right, and the reason that the map returned to the POST method raises an exception is the format.

### CODE REVIEW

File: src/playground/service.clj [hunk 1]
```diff
@@ -33,6 +33,12 @@
       (ring-resp/not-found "Entry not in DB")
       )))
 
+(defn insert-page [request]
+  (ring-resp/response (views/insert)))
+
+(defn insert-test [request]
```
Comment (manandearth) [inline src/playground/service.clj]: sorry this was a temporary test. I need to remove

**Comment by vemv** [comment]
Pushed some commits. Make sure to pull them. Note that these are unrelated to your goal, so normally they would go in a separate branch.

So sorry for the bad form ;p

**Comment by vemv** [comment]
Pushed a fix for your issue. You had to use :form-params rather than :params

Generally, always use the most specific possible "params key": `:query-params`, `:form-params`, `:path-params` are all preferable over plain `:params`.

Now, I don't know _exactly_ what was wrong about using params. If the issue persists (with a less trivial fix) another day, we can investigate.

File: src/playground/services/invoices/insert/endpoint.clj [hunk 2]
```diff
@@ -9,7 +9,7 @@
 
 (spec/def ::api (spec/keys :req-un [::amount]))
 
-(defn perform [{{:keys [amount]} :params :keys [db] :as request}]
+(defn perform [{{:keys [amount]} :form-params :keys [db] :as request}]
```
Comment (manandearth) [inline src/playground/services/invoices/insert/endpoint.clj:12]: Does `form-params` give back a map with a keyword then?

Comment (vemv) [reply src/playground/services/invoices/insert/endpoint.clj:12]: yes

File: dev/user.clj [hunk 3]
```diff
@@ -15,22 +15,35 @@
...
+   [formatting-stack.component]
    [modular.postgres]
+   [playground.server]
    #_ [background-processing.background-processor :as background-processor]
    #_ [background-processing.enqueuer :as enqueuer]
-   [playground.server]
    [playground.service]))
 
+;; ugly hack, will disappear when we use the newer project template
+(def vemv? (-> (System/getenv "USER") #{"vemv" "victor.valenzuela"}))
```
Comment (manandearth) [inline dev/user.clj:27]: How should I merge and pull the user part, here and in the modular configuration? should I just pull and change locally?

Comment (vemv) [reply dev/user.clj:27]: `git pull` and that's it

it will work for you, given the `if`s

Comment (manandearth) [reply dev/user.clj:27]: good

**Comment by manandearth** [comment]
Exception on jack-in

`Caused by: java.lang.RuntimeException: Unable to resolve var: clojure.main/root-cause in this context ...`

**Comment by manandearth** [comment]
https://gist.github.com/manandearth/3e39a36b2452a4c94baa164831229e1e

**Comment by vemv** [comment]
Add a explicit nrepl-middleware dependency to project.clj, which version has to match your Emacs'.

Determine that one by jacking-in in a different project, then `Cider -> About` or such

**Comment by vemv** [comment]
Also bump clojure to 1.10 in project.clj

**Pull request merged**